### PR TITLE
Deployment scenarios with k8s

### LINKED ISSUE #546: Add more deployment scenarios / details
### Describe the issue or suggestion

We are making changes to the [.NET Cloud page ](https://dotnet.microsoft.com/en-us/apps/cloud) to highlight Aspire come GA. One section will talk about deploying with Aspire and since there isn't enough room on the marketing page to cover the nuances of deployment, we are going to link to this doc for more details. 

#### Request 
- [ ] Update [Aspire Deployment page](https://learn.microsoft.com/en-us/dotnet/aspire/deployment/overview) to include information on how to deploy to Kubernetes using [Aspirate](https://github.com/prom3theu5/aspirational-manifests)
- [ ] Update [Aspire Deployment page](https://learn.microsoft.com/en-us/dotnet/aspire/deployment/overview) to include information on how to deploy to Kubernetes using kube manifests. 

@IEvangelist could you please add the Preview-5 label when you get a chance? Are there any other docs worth updating given these requested changes? 
@bradygaster 

---
[Associated WorkItem - 238742](https://dev.azure.com/msft-skilling/Content/_workitems/edit/238742)

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Deployment scenarios with k8s.

Fixes #546

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/deployment/overview.md](https://github.com/dotnet/docs-aspire/blob/d2840d57fad451d912e168a044d2df45f64594f3/docs/deployment/overview.md) | [.NET Aspire deployments](https://review.learn.microsoft.com/en-us/dotnet/aspire/deployment/overview?branch=pr-en-us-583) |

### CODE REVIEW

File: docs/deployment/overview.md [hunk 1]
```diff
@@ -2,27 +2,46 @@
...
 
 ### Use Application Insights for .NET Aspire telemetry
 
-.NET Aspire apps are designed to emit Telemetry using OpenTelemetry which uses a provider model. .NET Aspire Apps can direct their telemetry to Azure Monitor / Application Insights using the Azure Monitor telemetry distro. For more information, see [Application Insights](/azure/azure-monitor/app/app-insights-overview) for step-by-step instructions.
+.NET Aspire apps are designed to emit Telemetry using OpenTelemetry which uses a provider model. .NET Aspire Apps can direct their telemetry to Azure Monitor / Application Insights using the Azure Monitor telemetry distro. For more information, see [Use Application Insights for .NET Aspire telemetry](azure/application-insights.md) for step-by-step instructions.
+
+## Deploy to Kubernetes
+
+Kubernetes is a popular container orchestration platform that can run .NET Aspire applications. To deploy .NET Aspire apps to Kubernetes clusters, you'll need to map the .NET Aspire JSON manifest to a Kubernetes YAML manifest file. There are two ways to do this:
```
Comment (tdykstra) [inline docs/deployment/overview.md:32]:
```suggestion
Kubernetes is a popular container orchestration platform that can run .NET Aspire applications. To deploy .NET Aspire apps to Kubernetes clusters, you need to map the .NET Aspire JSON manifest to a Kubernetes YAML manifest file. There are two ways to do this: by using the Aspir8 project and by manually creating Kubernetes manifests.
```

File: docs/deployment/overview.md [hunk 2]
```diff
@@ -2,27 +2,46 @@
...
+
+### The Aspir8 project
+
+There's an open-source project named **Aspir8**, which handles the generation of deployment YAML based on the .NET Aspire app host manifest. The project outputs a .NET global tool, that can be used to perform a series of tasks, resulting in the generation of Kubernetes manifests:
```
Comment (tdykstra) [inline docs/deployment/overview.md:36]:
```suggestion
There's an open-source project named **Aspir8**, which handles the generation of deployment YAML based on the .NET Aspire app host manifest. The project outputs a .NET global tool that can be used to perform a series of tasks, resulting in the generation of Kubernetes manifests:
```

File: docs/deployment/overview.md [hunk 3]
```diff
@@ -2,27 +2,46 @@
...
+- `aspirate init`: Initializes the **Aspir8** project in the current directory.
+- `aspirate generate`: Generates Kubernetes manifests based on the .NET Aspire app host manifest.
+- `aspirate apply`: Applies the generated Kubernetes manifests to the Kubernetes cluster.
+- `aspirate destroy`: Deletes the resources created by the `apply` command.
+
+With these commands, you can build your apps, containerize them, and deploy them to Kubernetes clusters. For more information, see [Aspir8](https://prom3theu5.github.io/aspirational-manifests/getting-started.html).
+
+### Manually create Kubernetes manifests
+
+Alternatively, the Kubernetes manifests can be created manually. This involves more effort, and is a bit more time consuming. For more information, see [Deploy a .NET microservice to Kubernetes](/training/modules/dotnet-deploy-microservices-kubernetes/).
```
Comment (tdykstra) [inline docs/deployment/overview.md:47]:
```suggestion
Alternatively, the Kubernetes manifests can be created manually. This involves more effort and is more time consuming. For more information, see [Deploy a .NET microservice to Kubernetes](/training/modules/dotnet-deploy-microservices-kubernetes/).
```

**Review by tdykstra — APPROVED** [review]
LGTM, a few nit suggestions.

**Comment by claudiaregio** [comment]
@IEvangelist Aspir8 content seems reasonable to me but might be worth having @glennc, @bradygaster, or @DamianEdwards sign off on that part. Minor nit -> "There's an open-source project named.." didn't feel very formal for docs. I would maybe expect to see something like "Aspir8, an open-source project, handles the generation..." 

My question was around the Kubernetes manifests section -> It felt a bit odd to only direct visitors to a learn module. Would it be worth having _some_ instructions that would let experienced folks get moving and then have some way to say "if you're completely new to this we recommend the learn module"?

**Comment by IEvangelist** [comment]
> Aspir8, an open-source project, handles the generation

I'll update the first section as suggested, as for the manual part - happy to update it, but we literally don't have content on it anywhere else. That's all I was able to find, if we want to include any sort of bulleted list items, what are those steps?

**Comment by claudiaregio** [comment]
> > Aspir8, an open-source project, handles the generation
> 
> I'll update the first section as suggested, as for the manual part - happy to update it, but we literally don't have content on it anywhere else. That's all I was able to find, if we want to include any sort of bulleted list items, what are those steps?

@glennc @bradygaster Are there specific steps worth mentioning on these docs? Or is pointing to the learn module

**Comment by bradygaster** [comment]
@IEvangelist i was thinking about the page we have on deploying to container apps, the tab view? I think this page would pop more if we had a similar approach. Maybe the tabs would say "Deployment scenarios" and you'd have "AZD to ACA," "Kubernetes," and so on.

**Pull request merged**